src/language/loops.md

```diff
@@ -0,0 +1,137 @@
+---
```

Comment:
So there are no examples where a `for` loop uses a pattern to process the values obtained from the iterable?

```dart
for (var (x, y) in listOfPairs) {...}
```
Maybe that kind of example is in some other file?

Reply:
I didn't even think of that! I'll add it right under the non-pattern `for-in` example

Reply:
Very arbitrary, but I added:
```dart
for (var (x, y) in listOfPairs) {
  assert(x != y);
}
```